src/rpc/rpc.cc

```diff
@@ -134,7 +134,7 @@ namespace rpc {
 
   future<> connection::send_entry(outgoing_entry& d) noexcept {
     return futurize_invoke([this, &d] {
-      if (_propagate_timeout) {
+      if (d.buf.size && _propagate_timeout) {
```

Comment:
Shouldn't this be part of previous patch?

Reply:
No, because the previous patch is for the reader side, while this patch is for the writer side, and this line belongs to the latter.